thirdeye/thirdeye-pinot/src/main/resources/assets/javascript/DataService.js

```diff
@@ -147,8 +151,19 @@ DataService.prototype = {
       return this.getDataSynchronous(url);
     },
 
+    fetchMetricId(name) {
+      const url = "/data/data/metricId";
+      const data = {
+        name,
+      };
+
+      return this.getDataAsynchronous(url, data).then(([data]) => {
```

Comment:
this can be shorten`return this.getDataAsynchronous(url, data).then(([data]) => data.id);`